packages/snowpack/src/build/build-pipeline.ts

```diff
@@ -63,12 +68,37 @@ async function runPipelineLoadStep(
       const mainOutputExt = step.resolve.output[0];
       return {[mainOutputExt]: result};
     } else if (result && typeof result === 'object') {
+      result;
+
+      // handle source maps
+      Object.keys(result).forEach((ext) => {
+        const output = result[ext];
+        if (typeof output !== 'object' || !output.code) return;
+
+        if (output.map) {
+          result[ext + '.map'] = output.map;
```

Comment:
A bit of a weird decision that could use some input: @FredKSchott and I talked about 2 options for adding source map support for plugins:

**Option 1: plugins do everything**

Plugins can return an object of all the files they generate. That means they could do this:

```js
{
  '.js': // js code
  '.js.map': // js map
  '.css': // css code
  '.css.map': // css map
}
```

This is simple, but when plugins add the `//# sourceMappingURL =`, it sucks because they don’t know what Snowpack is going to name things (we’ve intentionally kept that simple). So every plugin would have some duplicated logic dealing with this.

**Option 2: plugins do less (shown here)**

Plugins instead can pass an object, inspired by Vue and Svelte’s compiler:

```js
{
  '.js': { code: /* js code */, map: /* js map */ },
  '.css': { code: /* css code */, map: /* css map */ },
}
```

This is nicer because plugins **DO NOT** have to do any weird `//# sourceMappingURL` code.

However, inside Snowpack, it does result in this weirdness, where we internally end up adding it to our output map just like Option 1. We just handle it for plugins.

I think I’m still leaning toward **Option 2** because it removes duplication from the plugins. But it can get weird where we treat `.map` as this sort of “secret meta“ file that’s invisible to the plugins.

I don’t have a specific question, but would welcome thoughts if anyone has any.